Source/JavaScriptCore/runtime/JSGenericTypedArrayViewPrototypeFunctions.h

```diff
@@ -746,32 +798,94 @@ ALWAYS_INLINE EncodedJSValue genericTypedArrayViewPrivateFuncSubarrayCreate(VM&
 
     size_t newByteOffset = thisObject->byteOffset() + offset * ViewClass::elementSize;
 
-    JSObject* defaultConstructor = globalObject->typedArrayConstructor(ViewClass::TypedArrayStorageType);
-    JSValue species = callFrame->uncheckedArgument(2);
-    if (species == defaultConstructor) {
+    scope.release();
+    return JSValue::encode(speciesConstruct(globalObject, thisObject, [&]() {
         Structure* structure = globalObject->typedArrayStructure(ViewClass::TypedArrayStorageType);
-
-        RELEASE_AND_RETURN(scope, JSValue::encode(ViewClass::create(
+        return ViewClass::create(
             globalObject, structure, WTFMove(arrayBuffer),
             thisObject->byteOffset() + offset * ViewClass::elementSize,
-            length)));
+            length);
+    }, [&](MarkedArgumentBuffer& args) {
+        args.append(vm.m_typedArrayController->toJS(globalObject, thisObject->globalObject(), arrayBuffer.get()));
+        args.append(jsNumber(newByteOffset));
+        args.append(jsNumber(length));
+        ASSERT(!args.hasOverflowed());
+    }));
+}
+
+template<typename ViewClass>
+static inline void validateIntegerIndex(JSGlobalObject* globalObject, ViewClass* thisObject, double index)
+{
+    auto scope = DECLARE_THROW_SCOPE(globalObject->vm());
+
+    if (UNLIKELY(thisObject->isDetached())) {
+        throwVMRangeError(globalObject, scope, typedArrayBufferHasBeenDetachedErrorMessage);
+        return;
     }
+    if (UNLIKELY(!isInteger(index))) {
+        throwVMRangeError(globalObject, scope, "index should be integer"_s);
+        return;
+    }
+    if (UNLIKELY(index == 0 && std::signbit(index))) {
+        throwVMRangeError(globalObject, scope, "index should not be negative zero"_s);
+        return;
+    }
+    if (UNLIKELY(index < 0 || index >= thisObject->length())) {
```

Comment:
nit:
```suggestion
    if (UNLIKELY(!thisObject->inBounds(index))) {
```

Reply:
Nice. Fixed.

Reply:
Ah, we need to have index < 0 check still.